### PR TITLE
Feature: package build script + Docker build script

### LINKED ISSUE #1944: Create a blessed Dockerfile for dbt
### Describe the feature
The people want a Dockerfile. Let's give the people what they want :)

We've historically pointed people to [this Dockerfile](https://github.com/davidgasquez/dbt-docker) (thanks @davidgasquez!) but it's worth creating a blessed version that we keep up to date with new dbt releases.

### Describe alternatives you've considered
- Deferring to open source versions of this Dockerfile
  - From Slack: "Blessed official docker images would be great for getting sign-off from corporate security."
  - I buy that :)

### PR DESCRIPTION
**Pull request by beckjake**
A package + docker image (+ dockerfile) building script.

Fixes #1944 

- Bumps the version (including commit)
- Builds pypi packages
- Uploads them to test pypi + prompts for enter/ctrl+c
- Uploads them to regular pypi
- Builds homebrew formula
  - it even passes `brew audit --strict`
- Overwrites the default homebrew package, if you want
- Builds a requirements.txt file
  - something to consider: Should this be done before the commit updating the version?
- Builds a docker container
  - Should this be done before the commit updating the version?
- Pushes a docker container
- Makes appropriate commits
- Has a bunch of flags to turn off parts of the build process

You can kind of imagine a world of Real CI where we build+push for each supported version of python, and then run the tests using those containers

Requirements:
  - a properly filled out .pypirc with credentials for testpypi and pypi
  - you must run it with python 3.8 for virtualenv/homebrew reasons: We need to use 3.8 so we get the right set of dependencies that homebrew will need.

Things this doesn't do (yet):
  - `git push`, so no github credentials required. Someday I would like it to run in the cloud though!
  - handle the homebrew bottle aspect
  - run in any CI stuff

### CODE REVIEW

**Comment by beckjake** [comment]
I'm going to rebase this back onto dev/barbara-gittings and add Dockerfile stuff to this PR directly.

File: scripts/build-dbt.py [hunk 1]
```diff
@@ -0,0 +1,829 @@
...
+            cwd=self.homebrew_path
+        )
+
+    @staticmethod
+    def run_tests(formula_path: Path):
+        path = os.path.normpath(formula_path)
+        run_command(['brew', 'uninstall', '--force', path])
+        run_command(['brew', 'install', path])
+        run_command(['brew', 'test', path])
+        run_command(['brew', 'audit', '--strict', path])
```
Comment (cmcarthur) [inline scripts/build-dbt.py]: idk if you've seen this yet, but be careful running these commands on your macbook. some of these brew commands will completely obliterate everything you have installed via brew

File: scripts/build-dbt.py [hunk 2]
```diff
@@ -0,0 +1,829 @@
...
+
+        return textwrap.dedent(
+            f'''\
+            FROM python:3.8.1-slim-buster
+
+            RUN apt-get update && \
+                apt-get dist-upgrade -y && \
+                apt-get install -y  --no-install-recommends \
+                    netcat curl git ssh  software-properties-common \
+                    make build-essential ca-certificates libpq-dev && \
```
Comment (cmcarthur) [inline scripts/build-dbt.py]: does this need netcat / curl / ssh?

**Comment by cmcarthur** [comment]
overall, this seems really great. there are a couple of comments in here that i'd like you to take a look at, but we should get this merged in after that, and treat any improvements incrementally.

File: scripts/build-dbt.py (hunk 1 above)
Comment (beckjake) [reply scripts/build-dbt.py]: Interesting, nothing bad has happened to me yet except postgres getting upgraded on me. I don't use brew for much except postgres. I guess until we get CI doing this part of it, that's just a risk I'll take.

File: scripts/build-dbt.py (hunk 2 above)
Comment (beckjake) [reply scripts/build-dbt.py]: After a very cursory test, everything appears ok without those!

**Comment by beckjake** [comment]
Implemented the PR feedback, added some fixes for small issues I discovered tonight (the newest commit), and rebased onto dev/barbara-gittings.

**Pull request merged**